Source/player.cpp

```diff
@@ -606,22 +606,18 @@ void __fastcall CreatePlayer(int pnum, char c)
 	plr[pnum]._pInfraFlag = 0;
 
 	if ( c == PC_WARRIOR ) {
-		plr[pnum]._pAblSpells[0] = 0x2000000;
-		plr[pnum]._pAblSpells[1] = 0;
+		plr[pnum]._pAblSpells64 = 0x2000000;

```

Comment:
Maybe we should use a macro to set/clear/check spells using their SPL_* enums.

Reply:
Agreed, should be `(__int64)1 << SPL_REPAIR`

Reply:
Did the following:
```
	if ( c == PC_WARRIOR ) {
		plr[pnum]._pAblSpells64 = (__int64)1 << SPL_REPAIR - 1;
	} else if ( c == PC_ROGUE ) {
		plr[pnum]._pAblSpells64 = (__int64)1 << SPL_DISARM - 1;
	} else if ( c == PC_SORCERER ) {
		plr[pnum]._pAblSpells64 = (__int64)1 << SPL_RECHARGE - 1;
	}
```

Reply:
That looks like it would trigger an operator precedence warning, should have brackets around (SPL_* - 1).

What I originally meant was maybe we need something like this:
```
#define SETSPELL(spellmask, spellID)  spellmask |= (unsigned __int64)1 << (spellID - 1)
#define CLEARSPELL(spellmask, spellID) spellmask &= ~((unsigned __int64)1 << (spellID - 1))
#define CHECKSPELL(spellmask, spellID (spellmask & ((unsigned __int64)1 << (spellID - 1)))
```

Reply:
Didnt' get a warning, but I have wrapped it just to be safe.

Reply:
Didn't make it into the commit that was merged? Visual Studio now emits the warnings when compiling nightly.